packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -1357,7 +1358,55 @@ test.serial('writeGranuleFromApi() writes a granule to PostgreSQL, DynamoDB, and
   t.true(await esGranulesClient.exists(granuleId));
 });
 
-test.serial('writeGranuleFromApi() writes a granule without an execution to PostgreSQL and DynamoDB.', async (t) => {
+test.serial('writeGranuleFromApi() given a partial granule updates only provided fields in PostgreSQL, DynamoDB, and Elasticsearch.', async (t) => {
+  const {
+    collectionCumulusId,
+    esClient,
+    esGranulesClient,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    knex,
+  } = t.context;
+
+  await writeGranuleFromApi({ ...granule }, knex, esClient, 'Create');
+
+  t.true(await granuleModel.exists({ granuleId }));
+  t.true(await granulePgModel.exists(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  ));
+  t.true(await esGranulesClient.exists(granuleId));
+
+  // Update existing granule with a partial granule object
+  const updatedGranule = {
+    granuleId,
+    collectionId: granule.collectionId,
+    cmrLink: 'updatedGranuled.com', // Only field we're changing
+    // FUTURE: In order to update a granule, the payload must include status and
+    // the status must be 'completed' or 'failed'
+    // if it's running or queued, it will try to insert the granule, not upsert
+    status: granule.status,
+  };
+
+  await writeGranuleFromApi({ ...updatedGranule }, knex, esClient, 'Update');
+
+  const dynamoGranule = await granuleModel.get({ granuleId });
```

Comment:
This approach is fine, however I think the test needs to validate the ES record matches as well.

Reply:
Also I'm adding it in 2857, but we should probably explicitly confirm that objects that have special behaviors (e.g. files, error?) handle correctly.     In actionable terms - we're validating specific fields (`cmrLInk`) but not the entire object against the test object.  We're only doing a deep equal here against pg vs dynamo (consistency) but not either of those to `updatedGranule` (correctness).

Reply:
Added and refactored the unit tests a bit to address this. It's over several commits and a lot has moved so it might be worth taking another look at the current state of https://github.com/nasa/cumulus/pull/3082/files#diff-e4b95ee0e3b92433a8e670895c1e8da32d083c3ca64e31761d4bb353cbb791a7